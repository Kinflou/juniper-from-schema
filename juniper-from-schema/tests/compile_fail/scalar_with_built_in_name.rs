#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../compile_pass/setup.rs");

juniper_from_schema::graphql_schema! {
    schema {
      query: Query
    }

    type Query {
      // The directive makes the return value `FieldResult<String>`
      // rather than the default `FieldResult<&String>`
      helloWorld(name: String!): String! @juniper(ownership: "owned")
    }

    scalar String
}

pub struct Query;

impl QueryFields for Query {
    fn field_hello_world(&self, executor: &Executor<Context>, name: String) -> FieldResult<String> {
        todo!()
    }
}
