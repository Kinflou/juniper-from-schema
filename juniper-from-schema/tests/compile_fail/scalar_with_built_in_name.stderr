error: proc macro panicked
  --> $DIR/scalar_with_built_in_name.rs:4:1
   |
4  | / juniper_from_schema::graphql_schema! {
5  | |     schema {
6  | |       query: Query
7  | |     }
...  |
15 | |     scalar String
16 | | }
   | |_^
   |
   = help: message:

           error: You cannot declare scalars with names matching a built-in
            --> schema:3:1
             |
           3 |    scalar String
             |    ^

           aborting due to previous error

error[E0405]: cannot find trait `QueryFields` in this scope
  --> $DIR/scalar_with_built_in_name.rs:20:6
   |
20 | impl QueryFields for Query {
   |      ^^^^^^^^^^^ not found in this scope
